src/sentry/tasks/data_export.py

```diff
@@ -1,8 +1,130 @@
+    if data_export.query_type == ExportQueryType.DISCOVER_V2:
+        # TODO(Leander): Implement logic for discover v2 CSVs
+        return
+    elif data_export.query_type == ExportQueryType.BILLING_REPORT:
+        # TODO(Leander): Implement logic for billing report CSVs
+        return
+    elif data_export.query_type == ExportQueryType.ISSUE_BY_TAG:
+        process_issue_by_tag(data_export.query_info)
+        return
+
+
+def process_discover_v2(data_export):
+    return
+
+
+def process_billing_report(data_export):
+    return
+
+
+def process_issue_by_tag(payload):
+    from sentry import tagstore
+    from sentry.models import EventUser, Group, Project, get_group_with_redirect
+
+    """
+    Convert tag payload to serialized JSON
+    Adapted from 'src/sentry/web/frontend/group_tag_export.py'
+    """
+    # Get the pertaining project
+    project = Project.objects.get(slug=payload["project_slug"])
+
+    # Get the pertaining issue
+    group, _ = get_group_with_redirect(
+        payload["group_id"], queryset=Group.objects.filter(project=project)
+    )
+
+    # Get the pertaining key
+    key = payload["key"]
+    lookup_key = u"sentry:{0}".format(key) if tagstore.is_reserved_key(key) else key
+
+    # If the key is the 'user' tag, attach the event user
+    def attach_eventuser(items):
+        users = EventUser.for_tags(group.project_id, [i.value for i in items])
+        for item in items:
+            item._eventuser = users.get(item.value)
+
+    # Create the fields/callback lists
+    if key == "user":
+        callbacks = [attach_eventuser]
+        fields = [
+            "value",
+            "id",
+            "email",
+            "username",
+            "ip_address",
+            "times_seen",
+            "last_seen",
+            "first_seen",
+        ]
+    else:
+        callbacks = []
+        fields = ["value", "times_seen", "last_seen", "first_seen"]
+
+    # Iterate endlessly through the GroupTagValues
+    iteration = 0
+    file_details = "{}-{}".format(payload["project_slug"], key)
+    file_name = get_file_name(ExportQueryType.ISSUE_BY_TAG_STR, file_details)
+
+    while True:
+        gtv_list = tagstore.get_group_tag_value_iter(
+            project_id=group.project_id,
+            group_id=group.id,
+            environment_id=None,
+            key=lookup_key,
+            callbacks=callbacks,
+            offset=SNUBA_MAX_RESUILTS * iteration,
+        )
+
+        gtv_list_raw = [serialize_issue_by_tag(key, item) for item in gtv_list]
+        if len(gtv_list_raw) == 0:
+            break
+        convert_to_csv(gtv_list_raw, fields, file_name, iteration == 0)
+        iteration += 1
+
+
+def get_file_name(type, custom_string, extension="csv"):
+    file_name = "{}-{}.{}".format(type, custom_string, extension)
+    return file_name
+
+
+def convert_to_csv(data, fields, file_name, include_header=False):
+    with open("/tmp/" + file_name, "a") as csvfile:
```

Comment:
Can we actually use https://docs.python.org/2/library/tempfile.html ?

Reply:
Thanks for the review! I actually just rebased off a working branch instead of cherry picking, this PR should only have the migration script. I'll update with SQL and everything